lexers/c.lua

```diff
@@ -11,6 +11,7 @@ lex:add_rule('keyword', lex:tag(lexer.KEYWORD, lex:word_match(lexer.KEYWORD)))
 
 -- Types.
 lex:add_rule('type', lex:tag(lexer.TYPE, lex:word_match(lexer.TYPE)))
+lex:add_rule('linux-types', lex:tag(lexer.TYPE .. '.linux', lex:word_match(lexer.TYPE .. '.linux')))
```

Comment:
I would prefer not to have domain-specific things in a standard C lexer. See comment here: https://github.com/orbitalquark/scintillua/discussions/124#discussioncomment-11643726

Sorry about that.